File: Frontend/TypeScript/token/token.ts
type TokenType = string

interface Token {
    Type: TokenType,
    Literal: string
}

const ILLEGAL = "ILLEGAL"
    , EOF = "EOF"

// Identifiers + literals
const IDENT = "IDENT"
    , INT = "INT"

// Operators
const ASSIGN = "="
    , PLUS = "+"
	, MINUS = "-"
	, BANG = "!"
	, ASTERISK = "*"
	, SLASH = "/"
	, LT = "<"
	, GT = ">"

// Delimiters
const COMMA = ","
    , SEMICOLON = ";"
    , LPAREN = "("
    , RPAREN = ")"
    , LBRACE = "{"
	, EQ = "=="
	, NOT_EQ = "!="


// Keywords
const FUNCTION="FUNCTION"
    , LET =  "LET"
    , TRUE = "TRUE"
    , FALSE = "FALSE"
    , IF = "IF"
    , ELSE = "ELSE"
    , RETURN = "RETURN"


let keywords = {
    "fn": FUNCTION,
    "let": LET,
    "true": TRUE,
    "false": FALSE,
    "if": IF,
    "else": ELSE,
    "return": RETURN,
}

File: Spyrals Backend/Go/token/token.go
package token

// Define token type and token struct
type TokenType string

type Token struct{
	Type TokenType
	Literal string
}

// Special characters and words
const (
	ILLEGAL = "ILLEGAL"
	EOF = "EOF"

	// Identifiers + literals
	IDENT = "IDENT" // identifiers aka variable names
	INT = "INT"

	// Operators
	ASSIGN = "="
	PLUS = "+"

	// Delimiters - chars that separate text strings
	COMMA = ","
	SEMICOLON = ";"

	LPAREN = "("
	RPAREN = ")"
	LBRACE = "{"
	RBRACE = "}"
)

var keywords = map[string] TokenType(
	"fn": FUNCTION,
	"let": LET,
)

func LookupIdent(ident string) TokenType {
	if tok, ok:= keywords[ident]; ok{
		return tok
	}
	return IDENT
}